src/features/history/effects.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import { getCurrentWindow } from '@electron/remote';
+import { AppThunk, logger } from '@nordicsemiconductor/pc-nrfconnect-shared';
+import { rename, rm, stat, writeFile } from 'fs/promises';
+import { dirname, join } from 'path';
+
+import {
+    MAXIMUM_MAX_NUMBER_OF_LINES,
+    MINIMUM_MAX_NUMER_OF_LINES,
+} from '../../app/store';
+import { TDispatch } from '../../app/thunk';
+import { RootState } from '../../appReducer';
+import HistoryBufferWrapper from './historyBuffer';
+import {
+    getMaximumNumberOfLinesInHistory,
+    getNumberOfLinesInHistory,
+    getPathToHistoryFile,
+    setMaxNumberOfLinesInHistoryFile,
+    setNumberOfLinesInHistory,
+} from './historySlice';
+
+let historyBuffer: Awaited<ReturnType<typeof HistoryBufferWrapper>>;
+
+export const initializeHistoryBuffer: AppThunk<RootState> = async (
+    dispatch,
+    getState
+) => {
+    if (historyBuffer != null) {
+        logger.debug(
+            'History Buffer is already initialized, and should only be initialized once.'
+        );
+        return;
+    }
+
+    const pathToHistoryFile = getPathToHistoryFile(getState());
+    historyBuffer = await HistoryBufferWrapper(pathToHistoryFile);
+
+    if (historyBuffer) {
+        dispatch(setNumberOfLinesInHistory(historyBuffer.getNumberOfLines()));
+        if (dispatch(validateNumberOfLinesInHistory)) {
+            logger.warn(
+                'History file is full, consider increasing the Max lines in history file, or cleaning it up, by using the Terminal Settings in the Side Panel.'
+            );
+        }
+    }
+
+    getCurrentWindow().on('focus', () => {
+        if (historyBuffer) {
+            logger.debug(
+                `The app is again in-focus and will need to read the content of ${pathToHistoryFile}, becuase it may not be in-sync with the (in-memory) history buffer.`
+            );
+            historyBuffer.refreshHistoryFromFile(pathToHistoryFile);
+            historyBuffer.redoHistoryMap();
+        }
+    });
+};
+
+const validateNumberOfLinesInHistory: AppThunk<RootState, boolean> = (
+    dispatch,
+    getState
+) => {
+    if (historyBuffer == null) {
+        logger.error(
+            'History Buffer is not initialized, this means that the app is not working as it should, and you should try to restart it. If the issue persists, please create a ticket on https://devzone.nordicsemi.com/'
```

Comment:
```suggestion
            'History Buffer is not initialized. Try to restart it or create a ticket on https://devzone.nordicsemi.com/ if the issue persists.'
```
Have been given the feedback of "don't write please" a lot recently so changed the wording a bit.
I also removed the "this means that..." because I don't think it is relevant to mention.

Reply:
I also extracted the message, as it is being reused multiple times.